packages/xenforo.php

```diff
@@ -301,6 +360,32 @@ protected function forumExport($ex) {
          where p.post_id <> t.first_post_id
             and message_state = 'visible'", $comment_Map);
 
+        $ex->exportTable('Media', "
+            select
+                a.attachment_id as MediaID,
+                ad.filename as Name,
+                concat('xf_attachments/', a.data_id, '-', replace(ad.filename, ' ', '_')) as Path,
+                ad.file_size as Size,
+                ad.user_id as InsertUserID,
+                from_unixtime(ad.upload_date) as DateInserted,
+                if(t.thread_id is null, a.content_id, t.thread_id) as ForeignID,
+                if(t.thread_id is null, 'comment', 'discussion') as ForeignTable,
+                ad.width as ImageWidth,
+                ad.height as ImageHeight,
+                concat('xf_attachments/', a.data_id, '-', replace(ad.filename, ' ', '_')) as ThumbPath
+            from
+                :_attachment a
+            left join
+                xf_post p on p.post_id = a.content_id
+            left join
+                :_thread t on t.first_post_id = a.content_id
+            join
+                :_attachment_data ad on ad.data_id = a.data_id
+            where
+                a.content_type = 'post'
+                and p.message_state <> 'deleted'
```

Comment:
I see you left join on xf_post, but we have a condition in the where with that table, is this handled properly if the left join returns nothing for some records ?

Reply:
If the left join returns nothing for a certain record it means "p.message_state" will be null. It will behave correctly since the condition verifies that "p.message_state" is not equal to "deleted".

Reply:
Why isn't that condition on the join instead ?

Reply:
In this case I think you are right. It would be better to add the condition directly on the join to reduce the number of matches instead of filtering in the "where", but the outcome will be the same.

Reply:
I will make the change.